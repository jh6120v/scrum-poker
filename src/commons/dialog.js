import React from 'react';
import PropTypes from 'prop-types';
import {
    CancelButton, ConfirmButton, DialogContent, DialogFooter, DialogShadow, DialogWrap
} from '../styles/dialog-style';

const Dialog = ({ message, onCancel, onConfirm }) => (message ? (
    <>
        <DialogShadow />
        <DialogWrap>
            <DialogContent>{message}</DialogContent>
            <DialogFooter>
                <CancelButton onClick={onCancel}>Cancel</CancelButton>
                <ConfirmButton onClick={onConfirm}>Confirm</ConfirmButton>
            </DialogFooter>
        </DialogWrap>
    </>
) : null);

Dialog.propsTypes = {
    message: PropTypes.string.isRequired,
    onCancel: PropTypes.func.isRequired,
    onConfirm: PropTypes.func.isRequired
};

export default Dialog;