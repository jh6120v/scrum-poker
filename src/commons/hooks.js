import React, { useState, useCallback } from 'react';
import {
    ConfirmButton, ModelContent, ModelFooter, ModelShadow, ModelWrap
} from '../styles/model-style';

const useModel = (message, confirm) => {
    const [isShown, setShown] = useState(false);
    const showModal = useCallback(() => setShown(true), []);
    const hideModal = useCallback(() => setShown(false), []);

    const ModelBox = () => (
        <div>
            <ModelWrap>
                <ModelContent>{message}</ModelContent>
                <ModelFooter>
                    <ConfirmButton onClick={confirm}>Confirm</ConfirmButton>
                </ModelFooter>
            </ModelWrap>
            <ModelShadow onClick={hideModal} />
        </div>
    );

    return {
        ModelBox,
        isShown,
        showModal,
        hideModal
    };
};

export {
    useModel
};
