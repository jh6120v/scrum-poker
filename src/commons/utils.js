import React, { Suspense } from 'react';
import { LazyLoad } from '../styles/common-style';

export const createActionCreator = (namespace) => (actionType) => {
    const type = `${namespace}/${actionType}`;
    const actionCreator = (payload = {}) => ({
        type,
        payload
    });

    actionCreator.type = type;
    Object.freeze(actionCreator);

    return actionCreator;
};

export const waitingRouteComponent = (Component) => () => (
    <Suspense fallback={<LazyLoad>Loading...</LazyLoad>}>
        <Component />
    </Suspense>
);

export const getRandom = (min, max) => Math.floor(Math.random() * (max - min + 1)) + min;
