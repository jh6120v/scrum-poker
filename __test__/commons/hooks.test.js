import React from 'react';
import { renderHook, act } from '@testing-library/react-hooks';
import { render } from '@testing-library/react';
import { useModel } from '../../src/commons/hooks';

describe('test custom hooks', () => {
    test('test useModel hooks', () => {
        const { result } = renderHook(() => useModel('test', () => {
            console.log('test');
        }));

        const { container } = render(<result.current.ModelBox />);

        expect(container.firstChild).toMatchSnapshot();

        act(() => {
            result.current.showModal();
        });

        expect(result.current.isShown).toBe(true);

        act(() => {
            result.current.hideModal();
        });

        expect(result.current.isShown).toBe(false);
    });
});
