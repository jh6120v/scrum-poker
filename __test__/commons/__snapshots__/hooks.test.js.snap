// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test custom hooks test useModel hooks 1`] = `
<div>
  <div
    class="sc-bwzfXH bWiNQO"
  >
    <div
      class="sc-htpNat IudBq"
    >
      test
    </div>
    <div
      class="sc-bxivhb ijgeFW"
    >
      <div
        class="sc-ifAKCX bSPprp"
      >
        Confirm
      </div>
    </div>
  </div>
  <div
    class="sc-bdVaJa fYNSRK"
  />
</div>
`;
